d2/src/main/java/com/linkedin/d2/balancer/clients/RetryClient.java

```diff
@@ -59,15 +75,31 @@
  */
 public class RetryClient extends D2ClientDelegator
 {
+  public static final long DEFAULT_UPDATE_INTERVAL_MS = TimeUnit.SECONDS.toMillis(5);
   private static final Logger LOG = LoggerFactory.getLogger(RetryClient.class);
 
+  private final Clock _clock;
+  private final LoadBalancer _balancer;
   private final int _limit;
+  private final long _updateIntervalMs;
 
-  public RetryClient(D2Client d2Client, int limit)
+  Map<String, CallTracker> _retryTrackerMap;
```

Comment:
I saw that when we invoke CallTracker.startCall, it's doing a lot more than just counting retry count and total call count, it's also counts the latency, outstanding latency and other stuff. Is this a little bit heavy for retry ratio? Especially that the CallTracker is used again in TrackerClient to track each server host's performance, this may defeat the purpose of CallTracker.

Reply:
Addressed. Use a separate light-weight CallTracker for RetryClient. Leave it to container to handle monitoring metrics.